test/functional/test_framework/test_framework.py

```diff
@@ -440,7 +466,7 @@ def sync_all(self, nodes=None, **kwargs):
 
     def _start_logging(self):
         # Add logger and logging handlers
-        self.log = logging.getLogger('TestFramework')
+        self.log = logging.getLogger('TestFramework.'+ self.options.tmpdir) # Assign new logger name to prevent temp path reuse.
```

Comment:
This isn't quite right. Since `logging.shutdown()` is called in `TestFramework.shutdown()`, no more logging should take place (see https://docs.python.org/3/library/logging.html#logging.shutdown).

A better way to do this would be to replace the `logging.shutdown()` call with:

```py
handlers = self.log.handlers
for h in handlers:
    h.flush()
    h.flush()
    self.log.removeHandler(h)
```

That way, when `setup` is called again, the old handlers will have been removed and the new handlers will be added to the `Logging` object.

Reply:
Thanks for this suggestion. Added to e163c98169e0405255a2e6acdbb679494b5fa42.